api/all.yaml

```diff
@@ -2138,6 +2159,38 @@ components:
       required:
         - full_name
         - version
+    GenerateTemporaryPathCredential:
+      type: object
+      properties:
+        url:
+          description: The URL of the storage path for which temporary credentials need to be generated.
+          type: string
+        operation:
+          $ref: '#/components/schemas/PathOperation'
+    # TODO: should we create a new schema for this? or just use temp table credential response?
+    GenerateTemporaryPathCredentialResponse:
```

Comment:
Yes, we should have a single `TemporaryCredentials` that's returned by all the temporary credential API calls